lib/internal/streams/legacy.js

```diff
@@ -111,4 +119,40 @@ function prependListener(emitter, event, fn) {
     emitter._events[event] = [fn, emitter._events[event]];
 }
 
+let defaultHighWaterMark = null;
+ObjectDefineProperty(Stream, 'defaultHighWaterMark', {
+  get () {
+    return defaultHighWaterMark || MathMax(Buffer.poolSize, 16 * 1024);
```

Comment:
This approach lgtm, also seems documented now.
0 is a correct value for hvm though but can't be set as default, is that intended?
If yes, just mentioning that in the documentation should be fine.